python/rpdk/java/templates/init/guided_aws/StubUpdateHandler.java

```diff
@@ -29,17 +29,45 @@
         // https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/blob/master/src/main/java/software/amazon/cloudformation/proxy/CallChain.java
 
         return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
-            // STEP 1 [first update/stabilize progress chain - required for resource update]
+
+            // STEP 1 [check if resource already exists]
+            // for more information -> https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract.html
+            // if target API does not support 'ResourceNotFoundException' then following check is required
```

Comment:
it seems we are checking for not found at multiple places. May be consolidate this to be in on place.

Reply:
we might be having same callGraph value cached in case you cross handler interaction. also I think it's already quite wordy if we refer in multiple places it will be confusing